### PR TITLE
test(chart): fix test to compatible with chartjs v4

### PR DESCRIPTION
**Pull request by Theeraphat-Sorasetsakul**
## Description

Migrate and update tests to compatible with Chartjs v4

Fixes # ([STG-201](https://jira.refinitiv.com/browse/STG-201))

## Type of change

- [x] Refactor (improves code without changing its functionality)

### CODE REVIEW

File: packages/elements/src/chart/index.ts [hunk 1]
```diff
@@ -205,6 +205,11 @@ export class Chart extends BasicElement {
    * @returns {ChartOptions} chart config with theme
    */
   protected get themableChartOption (): ChartOptions {
+    let boxHeight = Number(getComputedStyle(this).getPropertyValue('font-size').replace('px', ''));
+    if (this.config?.options?.plugins?.legend?.labels?.usePointStyle) {
+      boxHeight = this.cssVarAsNumber('--legend-key-box-width', '10') as number;
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/index.ts]: `--legend-key-box-width` value should be extracted only once. Let's refactor line 240 as well.

File: packages/elements/src/chart/__test__/mock-config.js [hunk 2]
```diff
@@ -786,129 +835,130 @@ let uc1 = {
...
+  return {
+    type: 'bar',
+    options: {
+      animation: {
+        duration: 0
+      },
+      maintainAspectRatio: false,
+      plugins: {
+        tooltips: {
+          callbacks: {
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/__test__/mock-config.js]: empty object as callbacks? This doesn't look right.

File: packages/elements/src/chart/__test__/mock-config.js [hunk 3]
```diff
@@ -786,129 +835,130 @@ let uc1 = {
...
-        bottomBound: '-0'
+      scales: {
+        y: {
+          position: 'left',
+          title: {
+            display: true,
+            text: 'Trades'
+          }
+        },
+        // xAxes: [{
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/__test__/mock-config.js]: not in use? Let's remove it.

File: packages/elements/src/chart/__test__/chart.test.js [hunk 4]
```diff
@@ -18,6 +27,7 @@ describe('chart/Chart', () => {
 
     beforeEach(async () => {
       el = await fixture('<ef-chart></ef-chart>');
+      config = createMock();
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/__test__/chart.test.js]:
```suggestion
      config = createMockConfig();
```

File: packages/elements/src/chart/__test__/chart.test.js [hunk 5]
```diff
@@ -18,6 +27,7 @@ describe('chart/Chart', () => {
...
     });
 
     it('DOM structure is correct', async () => {
```
Comment (wattachai-lseg) [inline packages/elements/src/chart/__test__/chart.test.js:33]: `expect()` doesn't return Promise. Let's remove `await`

**Review by wattachai-lseg — DISMISSED** [review]
just a few points

File: packages/elements/src/chart/index.ts (hunk 1 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/index.ts]: Updated by creating variable boxWidth.

File: packages/elements/src/chart/__test__/mock-config.js (hunk 2 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/__test__/mock-config.js]: I think it is an user config that was kept from the past. So I removed.

File: packages/elements/src/chart/__test__/mock-config.js (hunk 3 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/__test__/mock-config.js]: Removed.

File: packages/elements/src/chart/__test__/chart.test.js (hunk 4 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/__test__/chart.test.js]: Renamed to be createMockConfig

File: packages/elements/src/chart/__test__/chart.test.js (hunk 5 above)
Comment (Theeraphat-Sorasetsakul) [reply packages/elements/src/chart/__test__/chart.test.js:33]: Removed.

**Review by wattachai-lseg — APPROVED** [review]

**Pull request merged**